v2/client.go

```diff
@@ -366,26 +385,41 @@ func NewPointFrom(pt models.Point) *Point {
 func (c *client) Write(bp BatchPoints) error {
 	var b bytes.Buffer
 
+	var w io.Writer
+	if c.encoding == GzipEncoding {
+		w = gzip.NewWriter(&b)
+	} else {
+		w = &b
+	}
+
 	for _, p := range bp.Points() {
 		if p == nil {
 			continue
 		}
-		if _, err := b.WriteString(p.pt.PrecisionString(bp.Precision())); err != nil {
+		if _, err := io.WriteString(w, p.pt.PrecisionString(bp.Precision())); err != nil {
 			return err
 		}
 
-		if err := b.WriteByte('\n'); err != nil {
+		if _, err := w.Write([]byte{'\n'}); err != nil {
 			return err
 		}
 	}
 
+	// gzip writer should be closed to flush data into underlying buffer
+	if c, ok := w.(io.Closer); ok {
+		c.Close()
```

Comment:
This should check for an error. I think gzip is capable of returning errors during the close because it actually writes data to the stream. This shouldn't ever error but we should check it anyway.

Reply:
> This should check for an error. I think gzip is capable of returning errors during the close because it actually writes data to the stream. This shouldn't ever error but we should check it anyway.

done